### ISSUE TITLE
Compiler FE: Fix fuse_preactivation_batchnorm

### ISSUE THREAD

**Issue by seanshpark** [Samsung/ONE#5782]
`fuse_preactivation_batchnorm` option value testing fails with Segmentation fault.
- See `compiler/luci-pass-value-test/test.lst` comment.
- Let's fix this problem.
- Added: `luci-value-test` also fails

--> Segmentation fault is fixed by #5802

Value mismatch exist: we'll ignore this for now

**Comment by seanshpark** [comment]
While generating reference data in `eval_result_verifier.py`
`interpreter.set_tensor(input_details["index"], input_data)` line fails with `Segmentation fault (core dumped)`

`input_details["index"]` and `input_data` is a valid data : print() works ok.

**Comment by seanshpark** [comment]
Run verifier
```
cd build/debug/compiler/luci-pass-value-test

python ../../../../compiler/luci-pass-value-test/eval_result_verifier.py \
--driver ../../../../build/debug/compiler/luci-value-test/tester/luci_eval_tester \
--tflite ../../../../build/debug/compiler/common-artifacts/Net_Preactivation_BN_000.tflite \
--circle ../../../../build/debug/compiler/luci-pass-value-test/Net_Preactivation_BN_000.pass.circle
```

**Comment by seanshpark** [comment]
#5802 will fix segmentation fault. but still values differ.

**Comment by seanshpark** [comment]
@jinevening , from #4773 , I could see that you've tested with `ResNet50 V2` model.
have to done model execution result value test of `ResNet50 V2` before and after the Pass?

**Comment by seanshpark** [comment]
@jinevening , I've tried to enter `compiler/luci/pass/src/FuseActivationFunctionPass.cpp` with GDB,
so from `compiler/circle2circle-dredd-recipe-test/test.lst` file,
```
build/debug/compiler/circle2circle/circle2circle \
--fuse_preactivation_batchnorm \
build/debug/compiler/common-artifacts/Net_Preactivation_BN_000.circle \
build/debug/compiler/common-artifacts/Net_Preactivation_BN_000.2.circle
```
but this comment doesn't enter into `compiler/luci/pass/src/FuseActivationFunctionPass.cpp`

Seems `FuseActivationFunctionPass` Pass and `Net_Preactivation_BN_000` model doesn't seem to match the case.

Can you tell me how to check `FuseActivationFunctionPass` is working or not?

**Comment by seanshpark** [comment]
There are value diffs for this...
I made a smaller network to see it better and the values have difference

iuci-interpreter
```
[[[[-0.07438185 -0.03604344 -0.13506162]
   [-0.07438185 -0.03604344 -0.13506162]
   [-0.07438185 -0.03604344 -0.13506162]
   [-0.07438185 -0.03604344 -0.13506162]]
```
tflite interpreter
```
[[[[-0.07435987 -0.03606142 -0.13508081]
   [-0.07435987 -0.03606142 -0.13508081]
   [-0.07435987 -0.03606142 -0.13508081]
   [-0.07435987 -0.03606142 -0.13508081]]
```

**Comment by jinevening** [comment]
> have to done model execution result value test of ResNet50 V2 before and after the Pass?

Yes. I've checked the result in ResNet50 V2. There was a slight difference in values, but I and app part thought it is tolerable (it did not affect the classification result). A more serious problem was the results in quantized models. The value difference was magnified in quantized models, changing the classification results. So, we decided not to use this optimization in ResNet50 V2.

IMHO, it would be better not to test this pass in luci-value-pass-test. Or, it would be possible to make a looser condition for 'aggressive' optimizations and test this pass in that category.
